app/src/main/java/org/kontalk/util/MessageUtils.java

```diff
@@ -708,17 +735,29 @@ public static String toString(byte[] text) {
         return text;
     }
 
-    public static void convertSmileys(Editable input) {
+    public static boolean convertSmileys(Editable input) {
+        int i = 0;
         for (String key : sEmojiConverterMap.keySet()) {
-            replaceEditable(input, key, sEmojiConverterMap.get(key));
+            if (replaceEditable(input, key, sEmojiConverterMap.get(key))){
+                i++;
```

Comment:
rather than having a int that is incrementing, it seems all you really care about is if the flag was tripped or not in which case a bool here would do the job